File: NDL_Category/NDL_Category/CustomViewController.m
//
//  CustomViewController.m
//  NDL_Category
//
//  Created by ndl on 2018/1/18.
//  Copyright © 2018年 ndl. All rights reserved.
//

#import "CustomViewController.h"


@interface CustomViewController ()

@end

@implementation CustomViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = [UIColor cyanColor];
    // Do any additional setup after loading the view.
    
    UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
    btn.backgroundColor = [UIColor whiteColor];
    btn.frame = CGRectMake(0, 290, 100, 60);
    [btn setTitle:@"123456" forState:UIControlStateNormal];
    [self.view addSubview:btn];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: SwiftDemo/SwiftDemo/Common/CustomViewController.swift
//
//  CustomViewController.swift
//  SwiftDemo
//
//  Created by ndl on 2020/5/8.
//  Copyright © 2020 dzcx. All rights reserved.
//

import UIKit

class CustomViewController: UIViewController {
    
    var text: String!
    
    // 自定义指定构造器
//    init(with text: String) {
//        // 调用父类的designated方法
//        super.init(nibName: nil, bundle: nil)
//        self.text = text
//    }
//
//    required init?(coder: NSCoder) {
//        fatalError("init(coder:) has not been implemented")
//    }
    // 或者
    // 自定义便利构造器(便利构造器必须调用同类中定义的其他构造器，最终导致一个指定z构造器被调用)
    convenience init(with text: String) {
        self.init()// 实际调用过程self.init()->self.init(nibName: nil, bundle: nil)->super.init(nibName: nil, bundle: nil)
        self.text = text
    }
    
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    


}
